smurfs/src/actions/index.js

```diff
@@ -1,15 +1,59 @@
-/* 
-  Action Types Go Here!
-  Be sure to export each action type so you can pull it into your reducer
-*/
-
-/*
-  For this project you'll need at least 2 action creators for the main portion,
-   and 2 more for the stretch problem.
-   Be sure to include action types for each type of action creator. Also, be sure to mind
-     the "pending" states like, fetching, creating, updating and deleting.
-   C - addSmurf
-   R - getSmurfs
-   U - updateSmurf
-   D - deleteSmurf
-*/
+
+////////////////////////////////////////
+
+import axios from 'axios';
+
+////////////////////////////////////////
+
+export function getSmurfs(url) {
+
+  console.log('getSmurfs has triggered. ');
+  return (dispatch) => {
+    dispatch({type : "LOAD"});
+    axios.get(url)
+         .then( res => {
+           console.log("API get request succeeded.: ", res.data);
+           dispatch({type: "SUCCESS", payload: res.data})
+         })
+         .catch( err => {
+           console.log(err);
+         })
+  }
+}
+
+////////////////////////////////////////
+
+export function addSmurf(smurf) {
+  console.log('addSmurfs has triggered.');
+  return (dispatch) => {
+    dispatch({type: "LOAD"});
+    axios.post('https://localhost:3333/smurfs/',smurf)
+         .then( res => {
+            console.log("post successful ", res.data[res.data.length-1]);
+            dispatch({
+              type: "ADD", payload: res.data[res.data.length-1]
+            })
+         })
+         .catch( err => console.log('error', smurf));
+  }
+}
+
+////////////////////////////////////////
+
+export function remove(smurf) {
+  console.log('remove has triggered', smurf);
+
+  return (dispatch) => {
+    dispatch({type: "LOAD"});
+    axios.delete(`https://localhost:3000/smurfs/${smurf.id}`)
+         .then( res => {
+           console.log( 'delete success: ', res.data);
+           dispatch({type: "REMOVE", payload: smurf.id})
+          } )
+         .catch( err => console.log(err));
+  }
+} 
+
+////////////////////////////////////////
```

Comment:
I can see that you even did a stretch goal here with the remove action creator and it looks really good, everything looks ok. However I'd suggest adding the 'types' int its own variable, mainly because it's a convention in redux community and if someone else is looking at your code it would be easier for them having them saved into variables. It's also useful because VSCode has extensions that help you to autocomplete variable names, but mainly because if you make a typo you'll get an undefined error and you will know quickly what went wrong and where and how to fix it, rather than in this case, the way you did it, finding that typo wouldn't take too long, but noticing that your application is failing for a typo could take you even hours, this is a long paragraph for a suggestion but I wanna make clear you are not doing anything wrong here.